src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -150,14 +186,81 @@ private TypeSymbolWithAnnotations LvalueResultType
         /// <summary>
         /// Force the inference of the ResultType from LValueResultType.
         /// </summary>
-        private void UseLvalueOnly()
+        private void UseLvalueOnly(BoundExpression node)
         {
-            LvalueResultType = LvalueResultType;
+            var previousExpression = _currentExpression;
+            _currentExpression = node;
+            SetResult(LvalueResultType.ToTypeWithState(), LvalueResultType, isLvalue: true);
+            _currentExpression = previousExpression;
         }
 
-        private void SetResult(TypeWithState resultType, TypeSymbolWithAnnotations lvalueType)
+        private void SetInvalidResult() => SetResult(_invalidType, _invalidType.ToTypeSymbolWithAnnotations(), updateAnalyzedNullability: false);
+
+        private void SetResult(TypeWithState resultType, TypeSymbolWithAnnotations lvalueType, bool updateAnalyzedNullability = true, bool? isLvalue = null)
         {
             _visitResult = new VisitResult(resultType, lvalueType);
+            // We use the rvalue type for the public nullability because in most scenarios with differing lvalue and rvalue nullabilities,
```

Comment:
>// We use the rvalue type for the public nullability because in most scenarios with differing lvalue and rvalue nullabilities, [](start = 12, length = 126)

This confused me. I expected that the RValue would give use the state part of the NullabilityInfo, and the LValue would give us the annotation, unconditionally.